kpi/tests/test_assets.py

```diff
@@ -1002,3 +1003,40 @@ def test_asset_is_excluded_from_projects_list_flag(self):
         # Fetch the newly created asset and verify the flag is False
         owner_asset = Asset.objects.get(name='Breakfast')
         self.assertFalse(owner_asset.is_excluded_from_projects_list)
+
+    def test_asset_visibility_after_transfer(self):
+        """
+        Test to ensure that an asset shared with an organization owner remains
+        visible after the asset owner joins a different organization
+        """
+        # Step 1: Create an asset owned by an external user
+        self._create_asset_by_bob()
+        asset = Asset.objects.get(owner=self.external_user)
```

Comment:
I think it would be better to use the `uid` of the response to ensure we retrieve the correct asset.

```
response = self._create_asset_by_bob()
asset = Asset.objects.get(uid=response.data['uid'])
```